pkg/didcomm/protocol/didexchange/service.go

```diff
@@ -415,6 +418,90 @@ func (s *Service) AcceptExchangeRequest(connectionID, publicDID, label string) e
 	return s.accept(connectionID, publicDID, label, stateNameRequested, "accept exchange request")
 }
 
+// RespondTo this inbound invitation and return with the new connection record's ID.
+func (s *Service) RespondTo(i *OOBInvitation) (string, error) {
+	i.Type = oobMsgType
+
+	record, err := s.createConnectionRecord(i)
+	if err != nil {
+		return "", fmt.Errorf("failed to create connection record for oobinvitation : %w", err)
+	}
+
+	msg := &message{
+		Msg:           service.NewDIDCommMsgMap(i),
+		ThreadID:      i.ThreadID,
+		Options:       &options{},
+		NextStateName: stateNameInvited,
+		ConnRecord:    record,
+	}
+
+	err = s.storeEventTransientData(msg)
+	if err != nil {
+		return "", fmt.Errorf("failed to store transient event data : %w", err)
+	}
+
+	err = s.AcceptInvitation(record.ConnectionID, record.InvitationDID, record.TheirLabel)
```

Comment:
I believe we had 2 APIs before for Invitation (cc: @sudeshrshetty ). 
1. Handle Invitation -> get the invitation and save
2. AcceptInvitation -> User approves the invitation and sends exchange request.

Are we going to merge these 2 actions in new OutOfBand Invitation ? (just trying to understand)

Reply:
There's a `HandleInvitation()` in the didexchange client package: it is processed as an inbound didcomm message (creates connection record, sends didexchange request, etc)

There are two `AcceptInvitation()` - on the didexchange client and service - and they expect the connection record and all that transient data to exist prior to invocation. 

The out-of-band service does not store any data when it receives an inbound didcomm oob invitation.

Reply:
Yes, I understand. But my question was specifically about `RespondTo` function - why can't it just create connection record and store metadata (basically handleInvitation logic) without `Accepting` the Invitation ? Now it looks like we are combining these 2 actions.

Current BDD test -
https://github.com/hyperledger/aries-framework-go/blob/b0374299d8ac005db9c583334bd5c73f0e4a73b3/test/bdd/features/didexchange_e2e_sdk.feature#L22-L23

Reply:
It is currently reusing existing code. Plus "accepting the invitation" seemed like a step in the right direction, semantically speaking.

This PR is just a step in a transition. Completing the refactor on this service is a big task.

How do you see this happening?

Reply:
Oh ok. I guess, we will have clear picture of overall flow once we have OutOfBand-DIDExchange BDD tests.

Reply:
That's my next task